File: vortexdbg-api/src/main/java/com/vortexdbg/unix/struct/TimeVal64.java
package com.vortexdbg.unix.struct;

import com.vortexdbg.pointer.VortexdbgStructure;
import com.sun.jna.Pointer;

import java.util.Arrays;
import java.util.List;

public class TimeVal64 extends VortexdbgStructure {

    public TimeVal64(Pointer p) {
        super(p);
    }

    public long tv_sec;
    public long tv_usec;

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("tv_sec", "tv_usec");
    }

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/unix/struct/TimeVal64.kt
package com.vortexdbg.unix.struct

import com.vortexdbg.pointer.VortexdbgStructure
import com.sun.jna.Pointer

class TimeVal64(p: Pointer?) : VortexdbgStructure(p) {

    @JvmField
    var tv_sec: Long = 0
    @JvmField
    var tv_usec: Long = 0

    override fun getFieldOrder(): List<String> {
        return listOf("tv_sec", "tv_usec")
    }
}
